CHANGELOG.d/breaking_switch-to-es-modules.md

```diff
@@ -0,0 +1,28 @@
+* Switch from Common JS to ES modules
+
+  Previously, Purescript used Common JS for FFI declarations.
+
+  Before, FFI was declared like this...
+
+  ```javascript
+  const mymodule = require('mymodule')
+
+  exports.myvar = mymodule.myvar
+  ```
+
+  ...and will be changed to this...
+
+  ```javascript
+  import * as M from 'mymodule';
+  export const myvar = M.myvar
+  ```
+  ...or using the short version...
+
+  ```javascript
+  export { myvar } from 'mymodule';
+  ```
+
+* FFI is annotated with `/* #__PURE__ */` so that bundlers can perform DCE
+* If CJS is detected a `Warning` is emitted
```

Comment:
Relatedly, should we include this line, given that in a followup we are going to make this an error? I know that this is true in _this_ pull request, but we have immediate plans to change it.